src/views/Conversion.vue

```diff
@@ -171,10 +169,18 @@ export default class Conversion extends Mixins(
         }
 
         // parse firm conversion filing and original business snapshot into store
+        const firmSnapshot = await this.fetchFirmSnapshot()
         await this.parseConversionFiling(changeFiling, firmSnapshot)
       } else {
         // parse business data into store
-        await this.parseEntitySnapshot(firmSnapshot)
+        if (this.isConversionFiling) {
+          const conversionSnapshot = await this.fetchConversionSnapshot()
+          const conversionAddress = await this.fetchStoreConversionAddress()
+          await this.parseEntitySnapshot(conversionSnapshot, conversionAddress)
+        } else {
+          const firmSnapshot = await this.fetchFirmSnapshot()
+          await this.parseEntitySnapshot(firmSnapshot)
+        }
```

Comment:
And instead of using "404", we use this package:

![image](https://user-images.githubusercontent.com/10002889/170355712-12686b63-5870-42da-bd35-113842e22c02.png)

Reply:
Thanks for the suggestion. `fetchAddress()` will throw an error if it receives errors from BE, so I had to use a different function and change things around. I used `fetchConversionSnapshot()` instead of `fetchFirmSnapshot()` because it is much more complicated than just making an `axios` get call inside `fetchFirmSnapshot()`. I will use `NOT_FOUND`.

Reply:
My code on line 232 catches the error from `fetchAddresses()`.

It doesn't need to be complicated. Please simplify it. (Feel free to use my code above. If needed, add the extra checks for the error response message.)

Reply:
![image](https://user-images.githubusercontent.com/51458187/170365191-77449803-62d4-40fd-bd51-ecf3b953b16b.png)

![image](https://user-images.githubusercontent.com/51458187/170364791-bf97770a-b932-4c40-a447-b389f5147638.png)


Sorry Severin. I tried your code but this doesn't work because of the Promise reject.

Reply:
It's working for me locally.

Try without the space before `reason =>`.

Reply:
Thanks Severin. It's the space that causes the error. With this code, things can be much simpler now.